compiler/rustc_middle/src/ty/util.rs

```diff
@@ -876,6 +876,11 @@ impl<'tcx> TyCtxt<'tcx> {
     /// [public]: TyCtxt::is_private_dep
     /// [direct]: rustc_session::cstore::ExternCrate::is_direct
     pub fn is_user_visible_dep(self, key: CrateNum) -> bool {
+        // `#![rustc_private]` overrides defaults to make private dependencies usable.
+        if self.features().enabled(sym::rustc_private) {
```

Comment:
> However, this is a very internal feature, so it doesn't seem worth doing any additional filtering here.

Not really. There are a lot of external custom drivers that need to enable this feature.

Reply:
Are you suggesting that it is worth filtering here to only expose crates that are dependencies of `std`? Or just that the message is inaccurate.

Reply:
I think the override for rustc_private should be removed. Even with rustc_private enabled you generally shouldn't do something like `extern crate libc;` from the sysroot.